### PR TITLE
latex2html: init at 2016

### PR DESCRIPTION
**Pull request by yurrriq**
###### Motivation for this change
 
Add [latex2html](https://www.ctan.org/pkg/latex2html).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/latex2html/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+# Ported from Homebrew.
+# https://github.com/Homebrew/homebrew-core/blob/21834573f690407d34b0bbf4250b82ec38dda4d6/Formula/latex2html.rb
+
+stdenv.mkDerivation rec {
+  name = "latex2html-${version}";
+  version = "2016";
+
+  src = fetchurl {
+    url = http://mirrors.ctan.org/support/latex2html/latex2html-2016.tar.gz;
```
Comment (Mic92) [inline pkgs/tools/misc/latex2html/default.nix]: nitpick:

```
url = "http://mirrors.ctan.org/support/latex2html/latex2html-${version}.tar.gz;"
```

**Review by Mic92 — APPROVED** [review]

**Comment by yurrriq** [comment]
Good call! Thank you.

**Comment by yurrriq** [comment]
Force pushed with the amended `src.url`.

**Comment by Mic92** [comment]
The build contains spurious leftover to the build directory. Compiling a document to html results in the following error:

```
$ main.tex 2>&1 | grep -v "Unescaped"
Error: '/tmp/nix-build-latex2html-2016.drv-0' not usable as temporary directory.
$ grep /tmp/nix-build-latex2html-2016.drv-0 $(which latex2html)                                                                                                        
    $TEXEXPAND = "$PERL /tmp/nix-build-latex2html-2016.drv-0/latex2html-2016${dd}texexpand";
    $PSTOIMG   = "$PERL /tmp/nix-build-latex2html-2016.drv-0/latex2html-2016${dd}pstoimg";
    $ICONSERVER = L2hos->path2URL("/tmp/nix-build-latex2html-2016.drv-0/latex2html-2016${dd}icons");
    $RGBCOLORFILE = "/tmp/nix-build-latex2html-2016.drv-0/latex2html-2016${dd}styles${dd}rgb.txt";
    $CRAYOLAFILE = "/tmp/nix-build-latex2html-2016.drv-0/latex2html-2016${dd}styles${dd}crayola.txt";
```

**Comment by yurrriq** [comment]
@Mic92: I've ticked the "Allow edits from maintainers" box, so feel free to make any changes you see fit. I don't have the time for now.

**Comment by yurrriq** [comment]
<details>
  <summary>I've rebased and force pushed, attempting to address the error @Mic92 mentioned above.</summary>

```diff
diff --git a/pkgs/tools/misc/latex2html/default.nix b/pkgs/tools/misc/latex2html/default.nix
index 48976275cb..596a72ee5a 100644
--- a/pkgs/tools/misc/latex2html/default.nix
+++ b/pkgs/tools/misc/latex2html/default.nix
@@ -1,4 +1,5 @@
-{ stdenv, fetchurl, ghostscript, netpbm, texLive, perl }:
+{ stdenv, fetchurl, makeWrapper
+, ghostscript, netpbm, perl }:
 # TODO: withTex
 
 # Ported from Homebrew.
@@ -13,7 +14,9 @@ stdenv.mkDerivation rec {
     sha256 = "028k0ypbq94mlhydf1sbqlphlfl2fhmlzhgqq5jjzihfmccbq7db";
   };
 
-  buildInputs = [ ghostscript netpbm texLive perl ];
+  buildInputs = [ ghostscript netpbm perl ];
+
+  nativeBuildInputs = [ makeWrapper ];
 
   configurePhase = ''
     ./configure \
@@ -22,6 +25,12 @@ stdenv.mkDerivation rec {
       --with-texpath=$out/share/texmf/tex/latex/html
   '';
 
+  postInstall = ''
+    for p in $out/bin/{latex2html,pstoimg}; do \
+      wrapProgram $p --add-flags '--tmp="''${TMPDIR:-/tmp}"'
+    done
+  '';
+
   meta = with stdenv.lib; {
     description = "LaTeX-to-HTML translator";
     longDescription = ''
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 0a27f47a38..7600887e71 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -2646,9 +2646,7 @@ with pkgs;
 
   kindlegen = callPackage ../tools/typesetting/kindlegen { };
 
-  latex2html = callPackage ../tools/misc/latex2html {
-    texLive = texlive.combined.scheme-basic;
-  };
+  latex2html = callPackage ../tools/misc/latex2html { };
 
   ldapvi = callPackage ../tools/misc/ldapvi { };
 
```
</details>

**Comment by yurrriq** [comment]
/ping @Mic92, @fpletz, @pSub

This should be good to go now.

File: pkgs/tools/misc/latex2html/default.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      LaTeX (to the best of its abilities). It contains definitions from a wide
+      variety of classes and packages, and users may add further definitions by
+      writing Perl scripts that provide information about class/package
+      commands.
+    '';
+
+    homepage = "https://www.ctan.org/pkg/latex2html";
+
+    license = licenses.gpl2;
+    platforms = with platforms; linux ++ darwin;
```
Comment (yurrriq) [inline pkgs/tools/misc/latex2html/default.nix]: Should this be `platforms.all` or `inherit (perl.meta) platforms`?

**Pull request merged**